File: top_k_frequent/bucket_sort.py
def bucket_sort(nums, k):
    frequency_map = {}
    for num in nums:
        frequency_map[num] = frequency_map.get(num, 0) + 1
    
    buckets = [[] for _ in range(len(nums) + 1)]
    for num, freq in frequency_map.items():
        buckets[freq].append(num)
    
    result = []
    for i in range(len(buckets) - 1, 0, -1):
        for num in buckets[i]:
            result.append(num)
            if len(result) == k:
                print("Final Result:", result)
                return result
        
        
nums = [1, 1, 1, 2, 2, 3]
k = 2
print("The top", k, "frequent elements are:", bucket_sort(nums, k))        
    

File: top_k_frequent/bucket_sort.ts
export {};

function bucketSort(nums: number[], k: number): number[] {
  const frequencyMap = new Map<number, number>();
  for (const num of nums) {
    frequencyMap.set(num, (frequencyMap.get(num) || 0) + 1);
  }

  const buckets: number[][] = Array.from({ length: nums.length + 1 }, () => []);
  for (const [num, frequency] of frequencyMap) {
    buckets[frequency].push(num);
  }

  const result: number[] = [];
  for (let i = buckets.length - 1; i > 0; i--) {
    for (const num of buckets[i]) {
      result.push(num);
      if (result.length === k) return result;
    }
  }

  return result;
}

const nums = [1, 1, 1, 2, 2, 3, 5, 5, 5, 5, 6, 6, 7, 8, 8, 8, 8];
const k = 2;
const result = bucketSort(nums, k);
console.log("Result:", result);
